crates/rpc-replay/src/block_context.rs

```diff
@@ -25,11 +27,17 @@ pub fn build_block_context(
         block_number: BlockNumber(block.block_number),
         block_timestamp: BlockTimestamp(block.timestamp),
         sequencer_address,
+        // Inspiration taken from Papyrus:
+        // https://github.com/starkware-libs/sequencer/blob/7218aa1f7ca3fe21c0a2bede2570820939ffe069/crates/papyrus_execution/src/lib.rs#L363-L371
         gas_prices: GasPrices {
-            eth_l1_gas_price: felt_to_u128(&block.l1_gas_price.price_in_wei).try_into().unwrap(),
-            strk_l1_gas_price: felt_to_u128(&block.l1_gas_price.price_in_fri).try_into().unwrap(),
-            eth_l1_data_gas_price: felt_to_u128(&block.l1_data_gas_price.price_in_wei).try_into().unwrap(),
-            strk_l1_data_gas_price: felt_to_u128(&block.l1_data_gas_price.price_in_fri).try_into().unwrap(),
+            eth_l1_gas_price: felt_to_u128(&block.l1_gas_price.price_in_wei).try_into().unwrap_or(NonZeroU128::MIN),
+            strk_l1_gas_price: felt_to_u128(&block.l1_gas_price.price_in_fri).try_into().unwrap_or(NonZeroU128::MIN),
+            eth_l1_data_gas_price: felt_to_u128(&block.l1_data_gas_price.price_in_wei)
+                .try_into()
+                .unwrap_or(NonZeroU128::MIN),
+            strk_l1_data_gas_price: felt_to_u128(&block.l1_data_gas_price.price_in_fri)
+                .try_into()
+                .unwrap_or(NonZeroU128::MIN),
```

Comment:
I think we should be more careful with these checks because there are two cases that we are treating the same when we use `try_into().unwrap_or(NonZeroU128::MIN)`:

1. the intended case: `try_into()` errors because of the value `1`
2. the unintended case: `try_into()` errors because the felt overflows (is `> u128::MAX`)

The latter case should be handled more gracefully.

Reply:
Good catch 😉 I'll fix it asap